{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from concurrent.futures import ProcessPoolExecutor\n",
    "from pathlib import Path\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "import re\n",
    "import random\n",
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('recessive_overlap_100000.csv', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.loc[~df['genes'].isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loc</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>odds_ratio</th>\n",
       "      <th>number of genes in site</th>\n",
       "      <th>number of recessive genes in site</th>\n",
       "      <th>all genes</th>\n",
       "      <th>genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>('chr1', 160841995)</td>\n",
       "      <td>0.328210</td>\n",
       "      <td>2.738819</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>TRL-CAA6-1,HSPA7,RPS23P9,FCGR2C,FCGR3B,TRL-CAG...</td>\n",
       "      <td>FCGR3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>('chr10', 99809870)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>MIR6507,HPSE2</td>\n",
       "      <td>HPSE2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>('chr15', 54300516)</td>\n",
       "      <td>0.112490</td>\n",
       "      <td>12.326151</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>TEX9,MNS1,LOC105370832</td>\n",
       "      <td>MNS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>('chr15', 74463419)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SCAPER,MIR3713</td>\n",
       "      <td>SCAPER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>('chr16', 28978950)</td>\n",
       "      <td>0.212332</td>\n",
       "      <td>4.930209</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>EIF3C,LOC124903790,NPIPB7,CLN3,MIR6862-2,CDC37P2</td>\n",
       "      <td>CLN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>('chr16', 84393944)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>WWOX,LSM3P5</td>\n",
       "      <td>WWOX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>('chr19', 37905617)</td>\n",
       "      <td>0.272574</td>\n",
       "      <td>3.521458</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>EEF1A1P7,FFAR3,GPR42,LINC01531,CD22,MAG,FFAR1,...</td>\n",
       "      <td>MAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>('chr19', 57305157)</td>\n",
       "      <td>0.077425</td>\n",
       "      <td>4.931854</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>MBOAT7,LOC124904768,RPS9,LOC100420187,TSEN34,L...</td>\n",
       "      <td>MBOAT7,TSEN34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>('chr4', 154278324)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LRBA,RNA5SP168</td>\n",
       "      <td>LRBA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>('chr5', 156581766)</td>\n",
       "      <td>0.038997</td>\n",
       "      <td>inf</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>SGCD</td>\n",
       "      <td>SGCD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>('chr8', 15810562)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LOC124902060,TUSC3</td>\n",
       "      <td>TUSC3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     loc    pvalue  odds_ratio  number of genes in site  \\\n",
       "8    ('chr1', 160841995)  0.328210    2.738819                       10   \n",
       "17   ('chr10', 99809870)  0.076474   24.652720                        2   \n",
       "55   ('chr15', 54300516)  0.112490   12.326151                        3   \n",
       "57   ('chr15', 74463419)  0.076474   24.652720                        2   \n",
       "64   ('chr16', 28978950)  0.212332    4.930209                        6   \n",
       "67   ('chr16', 84393944)  0.076474   24.652720                        2   \n",
       "77   ('chr19', 37905617)  0.272574    3.521458                        8   \n",
       "82   ('chr19', 57305157)  0.077425    4.931854                       12   \n",
       "117  ('chr4', 154278324)  0.076474   24.652720                        2   \n",
       "129  ('chr5', 156581766)  0.038997         inf                        1   \n",
       "159   ('chr8', 15810562)  0.076474   24.652720                        2   \n",
       "\n",
       "     number of recessive genes in site  \\\n",
       "8                                    1   \n",
       "17                                   1   \n",
       "55                                   1   \n",
       "57                                   1   \n",
       "64                                   1   \n",
       "67                                   1   \n",
       "77                                   1   \n",
       "82                                   2   \n",
       "117                                  1   \n",
       "129                                  1   \n",
       "159                                  1   \n",
       "\n",
       "                                             all genes          genes  \n",
       "8    TRL-CAA6-1,HSPA7,RPS23P9,FCGR2C,FCGR3B,TRL-CAG...         FCGR3A  \n",
       "17                                       MIR6507,HPSE2          HPSE2  \n",
       "55                              TEX9,MNS1,LOC105370832           MNS1  \n",
       "57                                      SCAPER,MIR3713         SCAPER  \n",
       "64    EIF3C,LOC124903790,NPIPB7,CLN3,MIR6862-2,CDC37P2           CLN3  \n",
       "67                                         WWOX,LSM3P5           WWOX  \n",
       "77   EEF1A1P7,FFAR3,GPR42,LINC01531,CD22,MAG,FFAR1,...            MAG  \n",
       "82   MBOAT7,LOC124904768,RPS9,LOC100420187,TSEN34,L...  MBOAT7,TSEN34  \n",
       "117                                     LRBA,RNA5SP168           LRBA  \n",
       "129                                               SGCD           SGCD  \n",
       "159                                 LOC124902060,TUSC3          TUSC3  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gfab = 'hprc-v2.1-mc-chm13.gfab'\n",
    "gff3_file = \"chm13v2.0_RefSeq_Liftoff_v5.1.gff3\"\n",
    "fasta_file = \"/mnt/h700/omixlab/workflows/genome-project/resources/chm13v2.0.fa\"\n",
    "region = 'mhc1'\n",
    "cutpoints = 1\n",
    "graph_base = gfab.split('.')[0]\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "ref = 'chm13v2'\n",
    "chrom = \"chr6\"\n",
    "start = 29585529\n",
    "end = 32962695\n",
    "\n",
    "query_region = f'{ref}#{chrom}:{start}-{end}'\n",
    "\n",
    "workdir = f'{region}.{graph_base}.{chrom}#{start}_{end}.wd'\n",
    "viz_output = f'{workdir}/{region}.{cutpoints}.{graph_base}.gfa'\n",
    "connected_output = f'{workdir}/{region}.{cutpoints}.{graph_base}.walks.gfa'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'chm13v2#chr6:29585529-32962695'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "connected_command = f'gfabase sub {gfab} -o {connected_output} {query_region} --range --cutpoints {cutpoints} --view --no-sequences --connected'\n",
    "viz_command = f'gfabase sub {gfab} -o {viz_output} {query_region} --range --view --cutpoints {cutpoints}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_command(command_string):\n",
    "        print('running command')\n",
    "        cmd = command_string.split(' ')\n",
    "        print(cmd)\n",
    "        try:\n",
    "            result = subprocess.run(cmd, check=True, text=True)\n",
    "            return (cmd, None)\n",
    "        except subprocess.CalledProcessError as e:\n",
    "            print(\"error\")\n",
    "            return (cmd, None, e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def produce_plotable(gfab, gff3_file, region, cutpoints, graph_base, viz_output, connected_output, ref, chrom, start, end, query_region, workdir ):\n",
    "\n",
    "    Path(workdir).mkdir(parents=True, exist_ok=True)\n",
    "    connected_command = f'gfabase sub {gfab} -o {connected_output} {query_region} --range --cutpoints {cutpoints} --view --no-sequences --connected'\n",
    "    viz_command = f'gfabase sub {gfab} -o {viz_output} {query_region} --range --view --cutpoints {cutpoints}'\n",
    "    \n",
    "    print(connected_command)\n",
    "    \n",
    "    \n",
    "\n",
    "    os.system(connected_command)\n",
    "    os.system(viz_command)\n",
    "    # with ProcessPoolExecutor() as executor:\n",
    "    #     executor.map(run_command, [connected_command, viz_command])\n",
    "    def extract_genes_from_region(filename, chrom_filter, start_filter, end_filter):\n",
    "        genes = []\n",
    "        with open(filename, 'r') as gff:\n",
    "            for line in gff:\n",
    "                if not line.startswith(\"#\"):  # Ignore comment lines\n",
    "                    parts = line.strip().split(\"\\t\")\n",
    "                    chrom = parts[0]\n",
    "                    start = int(parts[3])\n",
    "                    end = int(parts[4])\n",
    "                    \n",
    "                    # Filter for the specific region\n",
    "                    if chrom == chrom_filter and parts[2] == \"gene\" and start_filter <= start and end_filter >= end:\n",
    "                        attributes = parts[8].split(\";\")\n",
    "                        gene_name = None\n",
    "                        for attribute in attributes:\n",
    "                            if attribute.startswith(\"gene_name=\"):\n",
    "                                gene_name = attribute.split(\"=\")[1]\n",
    "                                break\n",
    "                        if gene_name:\n",
    "                            genes.append((chrom,start, end, gene_name))\n",
    "        return genes\n",
    "\n",
    "    def extract_protein_coding_genes_from_region(filename, chrom_filter, start_filter, end_filter):\n",
    "        genes = []\n",
    "        with open(filename, 'r') as gff:\n",
    "            for line in gff:\n",
    "                if not line.startswith(\"#\"):  # Ignore comment lines\n",
    "                    parts = line.strip().split(\"\\t\")\n",
    "                    chrom = parts[0]\n",
    "                    start = int(parts[3])\n",
    "                    end = int(parts[4])\n",
    "                    \n",
    "                    # Filter for the specific region\n",
    "                    if chrom == chrom_filter and parts[2] == \"gene\" and start_filter <= start and end_filter >= end:\n",
    "                        attributes = parts[8].split(\";\")\n",
    "                        gene_name = None\n",
    "                        is_protein_coding = False\n",
    "                        for attribute in attributes:\n",
    "                            if attribute.startswith(\"gene_name=\"):\n",
    "                                gene_name = attribute.split(\"=\")[1]\n",
    "                            if attribute == \"gene_biotype=protein_coding\":\n",
    "                                is_protein_coding = True\n",
    "\n",
    "                        if gene_name and is_protein_coding:\n",
    "                            \n",
    "                            genes.append((chrom,start, end, gene_name))\n",
    "\n",
    "        return genes\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    all_genes_in_region = extract_genes_from_region(gff3_file, chrom, start, end) \n",
    "    pc_genes_in_region= extract_protein_coding_genes_from_region(gff3_file, chrom, start, end)\n",
    "    def extract_sequences_from_fasta(fasta_file, genes):\n",
    "        sequences = []\n",
    "        \n",
    "        fasta_dict = SeqIO.to_dict(SeqIO.parse(fasta_file, \"fasta\"))\n",
    "        for gene in genes:\n",
    "            chrom,start, end, gene_name = gene\n",
    "            gene_seq = fasta_dict['chr6'].seq[start-1:end]\n",
    "            record = (gene_name, str(gene_seq).upper())\n",
    "            sequences.append(record)\n",
    "\n",
    "        return sequences\n",
    "    all_gene_sequences = extract_sequences_from_fasta(fasta_file, all_genes_in_region)\n",
    "    pc_gene_sequences = extract_sequences_from_fasta(fasta_file, pc_genes_in_region)\n",
    "    def write_gene_fastas(fasta_dir, attr, gene_sequences):\n",
    "        Path(fasta_dir).mkdir(parents=True, exist_ok=True)\n",
    "        with open(f'{fasta_dir}/{region}_{attr}.fasta', 'w+') as f:\n",
    "            for gene, sequence in gene_sequences:\n",
    "                f.write(f'>{gene}\\n{sequence}\\n')\n",
    "        return f'{fasta_dir}/{region}_{attr}.fasta'\n",
    "    \n",
    "    \n",
    "    pc_fasta_file = write_gene_fastas(f'{workdir}/gene_fastas', 'protein_coding', pc_gene_sequences)\n",
    "    all_fasta_file = write_gene_fastas(f'{workdir}/gene_fastas', 'all', all_gene_sequences)\n",
    "    alignments_dir = f'{workdir}/gene_alignments'\n",
    "    pc_alignmnet_file = f'{alignments_dir}/aligned_pc_{region}.gaf'\n",
    "    all_alignmnet_file = f'{alignments_dir}/aligned_all_{region}.gaf'\n",
    "    Path(alignments_dir).mkdir(exist_ok=True, parents=True)\n",
    "    align_pc_genes = f'GraphAligner -g {viz_output} -f {pc_fasta_file} -t 32 -a {pc_alignmnet_file} -x vg --multimap-score-fraction 1'\n",
    "    align_all_genes= f'GraphAligner -g {viz_output} -f {all_fasta_file} -t 32 -a {all_alignmnet_file} -x vg --multimap-score-fraction 1'\n",
    "    \n",
    "    # with ProcessPoolExecutor() as executor:\n",
    "    #     executor.map(run_command, [align_pc_genes, align_all_genes])\n",
    "        \n",
    "    os.system(align_all_genes)\n",
    "    os.system(align_pc_genes)\n",
    "    \n",
    "    def get_segement_colors_from_alignment(alignment_file):\n",
    "        df = pd.read_csv(alignment_file, sep='\\t', header=None)\n",
    "        def random_color():\n",
    "            return \"#%06x\" % random.randint(0, 0xFFFFFF)\n",
    "        df['color'] = df.apply(lambda x: random_color(), axis=1)\n",
    "        \n",
    "        items=[]\n",
    "        for i, row in df.iterrows():\n",
    "\n",
    "            nodes = re.split(r'[<>]',row[5])\n",
    "            \n",
    "            for node in nodes:\n",
    "                if node != '':\n",
    "                    items.append((node, row['color']))\n",
    "        return pd.DataFrame(items)\n",
    "    \n",
    "    def get_segement_colors_from_alignment(alignment_file, color_files_dir):\n",
    "        df_all = pd.read_csv(alignment_file, sep='\\t', header=None)\n",
    "        df_all['index'] = df_all.groupby(0).cumcount() + 1\n",
    "        df = df_all[[0, 5]]\n",
    "        df = df.groupby(0).agg(lambda x: ''.join(x)).reset_index()\n",
    "        def random_color():\n",
    "            return \"#%06x\" % random.randint(0, 0xFFFFFF)\n",
    "\n",
    "        df['color'] = df.apply(lambda x: random_color(), axis=1)\n",
    "\n",
    "\n",
    "        items=[]\n",
    "        for i, row in df.iterrows():\n",
    "            gene = str(row[0])\n",
    "            nodes = re.split(r'[<>]',row[5])\n",
    "            with open(f'{color_files_dir}/{gene}.csv', 'w') as f:\n",
    "                \n",
    "            \n",
    "                for node in nodes:\n",
    "                    if node != '':\n",
    "                        f.write(f'{node}, {row[\"color\"]}\\n')\n",
    "                        items.append((node, row['color']))\n",
    "        return pd.DataFrame(items), df_all\n",
    "\n",
    "\n",
    "    color_files_dir = f'{workdir}/color_csvs'\n",
    "    Path(color_files_dir).mkdir(parents=True, exist_ok=True)\n",
    "    # get_segement_colors_from_alignment(pc_alignmnet_file).to_csv(f'{color_files_dir}/pc_colors.csv', header=None, index=None)\n",
    "    # get_segement_colors_from_alignment(all_alignmnet_file).to_csv(f'{color_files_dir}/all_colors.csv', header=None, index=None)\n",
    "    \n",
    "    pc_gene_colors, pc_genes_all = get_segement_colors_from_alignment(pc_alignmnet_file, color_files_dir)\n",
    "    pc_gene_colors.to_csv(f'{color_files_dir}/pc_colors.csv', header=None, index=None)\n",
    "    all_gene_colors, all_genes_all = get_segement_colors_from_alignment(all_alignmnet_file, color_files_dir)\n",
    "    all_gene_colors.to_csv(f'{color_files_dir}/all_colors.csv', header=None, index=None)\n",
    "    \n",
    "    viz_list = []\n",
    "    sample_walks = {} \n",
    "    with open(viz_output, 'r') as f:\n",
    "        lines = f.read().splitlines()\n",
    "        for line in lines:\n",
    "            if line[0] == 'S':\n",
    "                segment = line.split('\\t')[1]\n",
    "                viz_list.append(segment)\n",
    "                \n",
    "    with open(connected_output, 'r') as f:\n",
    "        lines = f.read().splitlines()\n",
    "        for line in lines:\n",
    "            if line[0] == 'W':\n",
    "                items = line.split('\\t')\n",
    "                nodes = re.split(r'[><]', items[6].strip())\n",
    "                if items[1] not in sample_walks:\n",
    "                    sample_walks[items[1]] = []\n",
    "                \n",
    "                sample_walks[items[1]].extend([node for node in nodes if node != '' ])\n",
    "                \n",
    "    genes_walks = {}\n",
    "\n",
    "    for i, row in pc_genes_all.iterrows():\n",
    "        gene = row[0]\n",
    "        gene_index = row['index']\n",
    "        path = re.split(r'[<>]',row[5])\n",
    "        s_i = f'{gene}_{gene_index}'\n",
    "        if gene not in genes_walks:\n",
    "            genes_walks[s_i] = []\n",
    "            \n",
    "        genes_walks[s_i].extend(path)\n",
    "        \n",
    "    sample_haps = {}\n",
    "    for k, v in sample_walks.items():\n",
    "        print(k)\n",
    "        if k not in sample_haps:\n",
    "            sample_haps[k] = []\n",
    "            \n",
    "        for kj, vj in genes_walks.items():\n",
    "            if len(set(v).intersection(set(vj))) > 1:\n",
    "                sample_haps[k].append(kj)\n",
    "                \n",
    "    hap_directory = f'{workdir}/sample_haps/'\n",
    "    Path(hap_directory).mkdir(parents=True, exist_ok=True)\n",
    "    with open(f'{hap_directory}/sample_haps', 'w') as f:\n",
    "        for i,haps in sample_haps.items():\n",
    "            f.write(f'{i} : {\",\".join(haps)}\\n')\n",
    "        \n",
    "               \n",
    "\n",
    "\n",
    "    for k,v in sample_walks.items():\n",
    "        sample_walks[k] = set(viz_list).intersection(set(v))\n",
    "    for k, v in sample_walks.items():\n",
    "        with open(f'{color_files_dir}/{k}.walks.csv', 'w') as f:\n",
    "            for item in v:\n",
    "                f.write(f'{item},#003366\\n')\n",
    "                \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('chm13.loc.bed', sep = '\\t', names=['chr', 'start', 'end'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "tasks = []\n",
    "for i, item in df.iterrows():\n",
    "    gfab = 'arp-v2.1-mc-chm13.gfab'\n",
    "    gff3_file = \"chm13v2.0_RefSeq_Liftoff_v5.1.gff3\"\n",
    "    fasta_file = \"/mnt/h700/omixlab/workflows/genome-project/resources/chm13v2.0.fa\"\n",
    "    region = i\n",
    "    cutpoints = 1\n",
    "    graph_base = gfab.split('.')[0]\n",
    "\n",
    "    chrom_item = item['chr']\n",
    "  \n",
    "    \n",
    "    start = int(item['start'])\n",
    "    end = int(item['end'])\n",
    "    chrom = chrom_item\n",
    "    ref = 'chm13v2'\n",
    "    \n",
    "    \n",
    "    \n",
    "\n",
    "    query_region = f'{ref}#{chrom}:{start}-{end}'\n",
    "\n",
    "    workdir = f'{region}.{graph_base}.{chrom}#{start}_{end}.wd'\n",
    "    viz_output = f'{workdir}/{region}.{cutpoints}.{graph_base}.gfa'\n",
    "    connected_output = f'{workdir}/{region}.{cutpoints}.{graph_base}.walks.gfa'\n",
    "    \n",
    "    \n",
    "    tasks.append((gfab, gff3_file, region, cutpoints, graph_base, viz_output, connected_output, ref, chrom, start, end, query_region, workdir))\n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('arp-v2.1-mc-chm13.gfab',\n",
       " 'chm13v2.0_RefSeq_Liftoff_v5.1.gff3',\n",
       " 0,\n",
       " 1,\n",
       " 'arp-v2',\n",
       " '0.arp-v2.chr12#64657050_68231266.wd/0.1.arp-v2.gfa',\n",
       " '0.arp-v2.chr12#64657050_68231266.wd/0.1.arp-v2.walks.gfa',\n",
       " 'chm13v2',\n",
       " 'chr12',\n",
       " 64657050,\n",
       " 68231266,\n",
       " 'chm13v2#chr12:64657050-68231266',\n",
       " '0.arp-v2.chr12#64657050_68231266.wd')"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tasks[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "tasks = []\n",
    "for i, item in df.iterrows():\n",
    "    gfab = 'hprc-v2.1-mc-chm13.gfab'\n",
    "    gff3_file = \"chm13v2.0_RefSeq_Liftoff_v5.1.gff3\"\n",
    "    fasta_file = \"/mnt/h700/omixlab/workflows/genome-project/resources/chm13v2.0.fa\"\n",
    "    region = i\n",
    "    cutpoints = 1\n",
    "    graph_base = gfab.split('.')[0]\n",
    "\n",
    "    chrom_item = item['loc'].split(',')[0][1:]\n",
    "    loc_item = item['loc'].split(',')[-1][:-1]\n",
    "    \n",
    "    start = int(loc_item)- 50000\n",
    "    end = int(loc_item) + 50000\n",
    "    chrom = chrom_item[1:-1]\n",
    "    ref = 'chm13v2'\n",
    "    \n",
    "    \n",
    "    \n",
    "\n",
    "    query_region = f'{ref}#{chrom}:{start}-{end}'\n",
    "\n",
    "    workdir = f'{region}.{graph_base}.{chrom}#{start}_{end}.wd'\n",
    "    viz_output = f'{workdir}/{region}.{cutpoints}.{graph_base}.gfa'\n",
    "    connected_output = f'{workdir}/{region}.{cutpoints}.{graph_base}.walks.gfa'\n",
    "    \n",
    "    \n",
    "    tasks.append((gfab, gff3_file, region, cutpoints, graph_base, viz_output, connected_output, ref, chrom, start, end, query_region, workdir))\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "from multiprocessing import Pool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loc</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>odds_ratio</th>\n",
       "      <th>number of genes in site</th>\n",
       "      <th>number of recessive genes in site</th>\n",
       "      <th>all genes</th>\n",
       "      <th>genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>('chr1', 160841995)</td>\n",
       "      <td>0.328210</td>\n",
       "      <td>2.738819</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>TRL-CAA6-1,HSPA7,RPS23P9,FCGR2C,FCGR3B,TRL-CAG...</td>\n",
       "      <td>FCGR3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>('chr10', 99809870)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>MIR6507,HPSE2</td>\n",
       "      <td>HPSE2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>('chr15', 54300516)</td>\n",
       "      <td>0.112490</td>\n",
       "      <td>12.326151</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>TEX9,MNS1,LOC105370832</td>\n",
       "      <td>MNS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>('chr15', 74463419)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>SCAPER,MIR3713</td>\n",
       "      <td>SCAPER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>('chr16', 28978950)</td>\n",
       "      <td>0.212332</td>\n",
       "      <td>4.930209</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>EIF3C,LOC124903790,NPIPB7,CLN3,MIR6862-2,CDC37P2</td>\n",
       "      <td>CLN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>('chr16', 84393944)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>WWOX,LSM3P5</td>\n",
       "      <td>WWOX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>('chr19', 37905617)</td>\n",
       "      <td>0.272574</td>\n",
       "      <td>3.521458</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>EEF1A1P7,FFAR3,GPR42,LINC01531,CD22,MAG,FFAR1,...</td>\n",
       "      <td>MAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>('chr19', 57305157)</td>\n",
       "      <td>0.077425</td>\n",
       "      <td>4.931854</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>MBOAT7,LOC124904768,RPS9,LOC100420187,TSEN34,L...</td>\n",
       "      <td>MBOAT7,TSEN34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>('chr4', 154278324)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LRBA,RNA5SP168</td>\n",
       "      <td>LRBA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>('chr5', 156581766)</td>\n",
       "      <td>0.038997</td>\n",
       "      <td>inf</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>SGCD</td>\n",
       "      <td>SGCD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>('chr8', 15810562)</td>\n",
       "      <td>0.076474</td>\n",
       "      <td>24.652720</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>LOC124902060,TUSC3</td>\n",
       "      <td>TUSC3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     loc    pvalue  odds_ratio  number of genes in site  \\\n",
       "8    ('chr1', 160841995)  0.328210    2.738819                       10   \n",
       "17   ('chr10', 99809870)  0.076474   24.652720                        2   \n",
       "55   ('chr15', 54300516)  0.112490   12.326151                        3   \n",
       "57   ('chr15', 74463419)  0.076474   24.652720                        2   \n",
       "64   ('chr16', 28978950)  0.212332    4.930209                        6   \n",
       "67   ('chr16', 84393944)  0.076474   24.652720                        2   \n",
       "77   ('chr19', 37905617)  0.272574    3.521458                        8   \n",
       "82   ('chr19', 57305157)  0.077425    4.931854                       12   \n",
       "117  ('chr4', 154278324)  0.076474   24.652720                        2   \n",
       "129  ('chr5', 156581766)  0.038997         inf                        1   \n",
       "159   ('chr8', 15810562)  0.076474   24.652720                        2   \n",
       "\n",
       "     number of recessive genes in site  \\\n",
       "8                                    1   \n",
       "17                                   1   \n",
       "55                                   1   \n",
       "57                                   1   \n",
       "64                                   1   \n",
       "67                                   1   \n",
       "77                                   1   \n",
       "82                                   2   \n",
       "117                                  1   \n",
       "129                                  1   \n",
       "159                                  1   \n",
       "\n",
       "                                             all genes          genes  \n",
       "8    TRL-CAA6-1,HSPA7,RPS23P9,FCGR2C,FCGR3B,TRL-CAG...         FCGR3A  \n",
       "17                                       MIR6507,HPSE2          HPSE2  \n",
       "55                              TEX9,MNS1,LOC105370832           MNS1  \n",
       "57                                      SCAPER,MIR3713         SCAPER  \n",
       "64    EIF3C,LOC124903790,NPIPB7,CLN3,MIR6862-2,CDC37P2           CLN3  \n",
       "67                                         WWOX,LSM3P5           WWOX  \n",
       "77   EEF1A1P7,FFAR3,GPR42,LINC01531,CD22,MAG,FFAR1,...            MAG  \n",
       "82   MBOAT7,LOC124904768,RPS9,LOC100420187,TSEN34,L...  MBOAT7,TSEN34  \n",
       "117                                     LRBA,RNA5SP168           LRBA  \n",
       "129                                               SGCD           SGCD  \n",
       "159                                 LOC124902060,TUSC3          TUSC3  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 0.arp-v2.chr12#64657050_68231266.wd/0.1.arp-v2.walks.gfa chm13v2#chr12:64657050-68231266 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 1.arp-v2.chr15#28411284_29947360.wd/1.1.arp-v2.walks.gfa chm13v2#chr15:28411284-29947360 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 2.arp-v2.chr15#71986221_73551280.wd/2.1.arp-v2.walks.gfa chm13v2#chr15:71986221-73551280 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 11.arp-v2.chr22#21976303_23803198.wd/11.1.arp-v2.walks.gfa chm13v2#chr22:21976303-23803198 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 4.arp-v2.chr16#29877571_30473116.wd/4.1.arp-v2.walks.gfa chm13v2#chr16:29877571-30473116 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 3.arp-v2.chr15#96579126_99750708.wd/3.1.arp-v2.walks.gfa chm13v2#chr15:96579126-99750708 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 5.arp-v2.chr16#14891230_16405045.wd/5.1.arp-v2.walks.gfa chm13v2#chr16:14891230-16405045 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 6.arp-v2.chr16#14891230_16405045.wd/6.1.arp-v2.walks.gfa chm13v2#chr16:14891230-16405045 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 24.arp-v2.chrY#13139492_13951220.wd/24.1.arp-v2.walks.gfa chm13v2#chrY:13139492-13951220 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 7.arp-v2.chr17#46489484_47078704.wd/7.1.arp-v2.walks.gfa chm13v2#chr17:46489484-47078704 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 9.arp-v2.chr1#146396099_147746605.wd/9.1.arp-v2.walks.gfa chm13v2#chr1:146396099-147746605 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 8.arp-v2.chr1#146396099_147746605.wd/8.1.arp-v2.walks.gfa chm13v2#chr1:146396099-147746605 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 37.arp-v2.chr17#37444976_38843267.wd/37.1.arp-v2.walks.gfa chm13v2#chr17:37444976-38843267 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 10.arp-v2.chr1#144803785_145165379.wd/10.1.arp-v2.walks.gfa chm13v2#chr1:144803785-145165379 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 12.arp-v2.chr2#59063565_61598724.wd/12.1.arp-v2.walks.gfa chm13v2#chr2:59063565-61598724 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 13.arp-v2.chr2#44188661_44367747.wd/13.1.arp-v2.walks.gfa chm13v2#chr2:44188661-44367747 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 15.arp-v2.chr2#239539540_239891533.wd/15.1.arp-v2.walks.gfa chm13v2#chr2:239539540-239891533 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 16.arp-v2.chr3#198719061_200350356.wd/16.1.arp-v2.walks.gfa chm13v2#chr3:198719061-200350356 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 14.arp-v2.chr2#196544429_204824491.wd/14.1.arp-v2.walks.gfa chm13v2#chr2:196544429-204824491 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 17.arp-v2.chr3#198719061_200350356.wd/17.1.arp-v2.walks.gfa chm13v2#chr3:198719061-200350356 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 18.arp-v2.chr7#74530623_75930620.wd/18.1.arp-v2.walks.gfa chm13v2#chr7:74530623-75930620 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 20.arp-v2.chr8#7833177_11501099.wd/20.1.arp-v2.walks.gfa chm13v2#chr8:7833177-11501099 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 19.arp-v2.chr8#7833177_11501099.wd/19.1.arp-v2.walks.gfa chm13v2#chr8:7833177-11501099 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 21.arp-v2.chr8#76743643_77283136.wd/21.1.arp-v2.walks.gfa chm13v2#chr8:76743643-77283136 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 22.arp-v2.chr9#149870673_150093988.wd/22.1.arp-v2.walks.gfa chm13v2#chr9:149870673-150093988 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 23.arp-v2.chr16#3490_789083.wd/23.1.arp-v2.walks.gfa chm13v2#chr16:3490-789083 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 28.arp-v2.chr21#24238621_24529258.wd/28.1.arp-v2.walks.gfa chm13v2#chr21:24238621-24529258 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 27.arp-v2.chr17#14102393_15473806.wd/27.1.arp-v2.walks.gfa chm13v2#chr17:14102393-15473806 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 29.arp-v2.chr5#113218068_113356772.wd/29.1.arp-v2.walks.gfa chm13v2#chr5:113218068-113356772 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 30.arp-v2.chr17#14102393_15473806.wd/30.1.arp-v2.walks.gfa chm13v2#chr17:14102393-15473806 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 31.arp-v2.chrX#630829_742852.wd/31.1.arp-v2.walks.gfa chm13v2#chrX:630829-742852 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 33.arp-v2.chr17#31725057_32882176.wd/33.1.arp-v2.walks.gfa chm13v2#chr17:31725057-32882176 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 34.arp-v2.chrX#102223467_102239584.wd/34.1.arp-v2.walks.gfa chm13v2#chrX:102223467-102239584 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 32.arp-v2.chrX#332539_632828.wd/32.1.arp-v2.walks.gfa chm13v2#chrX:332539-632828 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 35.arp-v2.chr11#44128586_46186902.wd/35.1.arp-v2.walks.gfa chm13v2#chr11:44128586-46186902 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 36.arp-v2.chr15#21111068_25935855.wd/36.1.arp-v2.walks.gfa chm13v2#chr15:21111068-25935855 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 40.arp-v2.chr5#176841220_178167392.wd/40.1.arp-v2.walks.gfa chm13v2#chr5:176841220-178167392 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 42.arp-v2.chrX#6091130_7718796.wd/42.1.arp-v2.walks.gfa chm13v2#chrX:6091130-7718796 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 38.arp-v2.chr16#21780281_22282000.wd/38.1.arp-v2.walks.gfa chm13v2#chr16:21780281-22282000 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 39.arp-v2.chr16#3752324_3907325.wd/39.1.arp-v2.walks.gfa chm13v2#chr16:3752324-3907325 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 44.arp-v2.chr7#74530623_75930620.wd/44.1.arp-v2.walks.gfa chm13v2#chr7:74530623-75930620 --range --cutpoints 1 --view --no-sequences --connectedgfabase sub arp-v2.1-mc-chm13.gfab -o 43.arp-v2.chr11#31920604_32571026.wd/43.1.arp-v2.walks.gfa chm13v2#chr11:31920604-32571026 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 45.arp-v2.chr4#1564721_2107304.wd/45.1.arp-v2.walks.gfa chm13v2#chr4:1564721-2107304 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 46.arp-v2.chrX#52660456_52942675.wd/46.1.arp-v2.walks.gfa chm13v2#chrX:52660456-52942675 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 41.arp-v2.chr7#97924409_97945534.wd/41.1.arp-v2.walks.gfa chm13v2#chr7:97924409-97945534 --range --cutpoints 1 --view --no-sequences --connected\n",
      "\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 47.arp-v2.chrX#152295493_152371389.wd/47.1.arp-v2.walks.gfa chm13v2#chrX:152295493-152371389 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 26.arp-v2.chr15#21111068_25935855.wd/26.1.arp-v2.walks.gfa chm13v2#chr15:21111068-25935855 --range --cutpoints 1 --view --no-sequences --connected\n",
      "gfabase sub arp-v2.1-mc-chm13.gfab -o 25.arp-v2.chr5#127296578_127357028.wd/25.1.arp-v2.walks.gfa chm13v2#chr5:127296578-127357028 --range --cutpoints 1 --view --no-sequences --connected\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][3.4s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][3.5s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][3.6s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][3.7s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][3.8s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][3.8s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][3.8s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][3.9s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][4.2s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][4.2s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][4.3s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][4.4s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.1s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.0s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.3s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.3s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.3s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.6s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.6s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.6s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.6s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.7s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][5.8s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.0s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.1s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.2s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.2s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.3s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.4s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.4s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.7s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.7s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.7s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.9s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][6.9s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][7.1s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][7.1s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][7.1s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][7.1s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][7.4s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][7.9s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][8.2s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][8.2s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][8.6[s] --connected overriding --cutpoints\u001b[\n",
      "33mWARN\u001b[0m][8.6s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][8.7s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][8.7s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][8.8s] --connected overriding --cutpoints\n",
      "[\u001b[33mWARN\u001b[0m][1.1s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 24.arp-v2.chrY#13139492_13951220.wd/24.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 24.arp-v2.chrY#13139492_13951220.wd/gene_alignments/aligned_all_24.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 18 (419963bp)\n",
      "Seeds found: 54163\n",
      "Seeds extended: 1712\n",
      "Reads with a seed: 17 (419693bp)\n",
      "Reads with an alignment: 17 (110694bp)\n",
      "Alignments: 1085 (110827bp) (627 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 24.arp-v2.chrY#13139492_13951220.wd/24.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 24.arp-v2.chrY#13139492_13951220.wd/gene_alignments/aligned_pc_24.gaf\n",
      "Align\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment finished\n",
      "Input reads: 2 (176090bp)\n",
      "Seeds found: 27369\n",
      "Seeds extended: 782\n",
      "Reads with a seed: 2 (176090bp)\n",
      "Reads with an alignment: 2 (56744bp)\n",
      "Alignments: 470 (57536bp) (312 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "chm13v2\n",
      "father\n",
      "gp001\n",
      "gp002\n",
      "gp003\n",
      "gp005\n",
      "gp006\n",
      "gp009\n",
      "gp010\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.3s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp011\n",
      "gp012\n",
      "gp013\n",
      "gp017\n",
      "gp023\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][1.3s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.1s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp025\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.4s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.1s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.1s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.8s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.0s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp026\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][1.0s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.8s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.8s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.8s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 28.arp-v2.chr21#24238621_24529258.wd/28.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 28.arp-v2.chr21#24238621_24529258.wd/gene_alignments/aligned_all_28.gaf\n",
      "Align\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.6s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment finished\n",
      "Input reads: 3 (325218bp)\n",
      "Seeds found: 36541\n",
      "Seeds extended: 876\n",
      "Reads with a seed: 2 (325116bp)\n",
      "Reads with an alignment: 2 (75968bp)\n",
      "Alignments: 547 (76629bp) (329 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 28.arp-v2.chr21#24238621_24529258.wd/28.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 28.arp-v2.chr21#24238621_24529258.wd/gene_alignments/aligned_pc_28.gaf\n",
      "Align\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment finished\n",
      "Input reads: 1 (290637bp)\n",
      "Seeds found: 33355\n",
      "Seeds extended: 799\n",
      "Reads with a seed: 1 (290637bp)\n",
      "Reads with an alignment: 1 (69290bp)\n",
      "Alignments: 493 (69950bp) (306 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "gp027\n",
      "gp033\n",
      "gp034\n",
      "gp036\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.5s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.1s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.3s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][1.6s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.6s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.2s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp038\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.5s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chm13v2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.8s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][1.8s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp039\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][4.5s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][2.5s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 11.arp-v2.chr22#21976303_23803198.wd/11.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 11.arp-v2.chr22#21976303_23803198.wd/gene_alignments/aligned_all_11.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 147 (1864450bp)\n",
      "Seeds found: 186798\n",
      "Seeds extended: 7480\n",
      "Reads with a seed: 123 (1856479bp)\n",
      "Reads with an alignment: 123 (433547bp)\n",
      "Alignments: 5602 (440517bp) (1878 additional alignments discarded)\n",
      "End-to-end alignments: 3 (2915bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 11.arp-v2.chr22#21976303_23803198.wd/11.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "gp041\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 11.arp-v2.chr22#21976303_23803198.wd/gene_alignments/aligned_pc_11.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 18 (640780bp)\n",
      "Seeds found: 75725\n",
      "Seeds extended: 2650\n",
      "Reads with a seed: 18 (640780bp)\n",
      "Reads with an alignment: 18 (156176bp)\n",
      "Alignments: 1925 (158154bp) (725 additional alignments discarded)\n",
      "End-to-end alignments: 1 (1934bp)\n",
      "hg38_no_alt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 32.arp-v2.chrX#332539_632828.wd/32.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 32.arp-v2.chrX#332539_632828.wd/gene_alignments/aligned_all_32.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 3 (35454bp)\n",
      "Seeds found: 4733\n",
      "Seeds extended: 53\n",
      "Reads with a seed: 1 (34976bp)\n",
      "Reads with an alignment: 1 (3382bp)\n",
      "Alignments: 42 (3398bp) (11 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "son\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 32.arp-v2.chrX#332539_632828.wd/32.1.arp-v2.gfa\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 31.arp-v2.chrX#630829_742852.wd/31.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Build alignment graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 31.arp-v2.chrX#630829_742852.wd/gene_alignments/aligned_all_31.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 31.arp-v2.chrX#630829_742852.wd/31.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 34.arp-v2.chrX#102223467_102239584.wd/34.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 31.arp-v2.chrX#630829_742852.wd/gene_alignments/aligned_pc_31.gaf\n",
      "Align\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 34.arp-v2.chrX#102223467_102239584.wd/gene_alignments/aligned_all_34.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 34.arp-v2.chrX#102223467_102239584.wd/34.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 32.arp-v2.chrX#332539_632828.wd/gene_alignments/aligned_pc_32.gaf\n",
      "Align\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 34.arp-v2.chrX#102223467_102239584.wd/gene_alignments/aligned_pc_34.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Alignment finished\n",
      "Input reads: 1 (34976bp)\n",
      "Seeds found: 4733\n",
      "Seeds extended: 53\n",
      "Reads with a seed: 1 (34976bp)\n",
      "Reads with an alignment: 1 (3382bp)\n",
      "Alignments: 42 (3398bp) (11 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 42.arp-v2.chrX#6091130_7718796.wd/42.1.arp-v2.gfa\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 46.arp-v2.chrX#52660456_52942675.wd/46.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 46.arp-v2.chrX#52660456_52942675.wd/gene_alignments/aligned_all_46.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 8 (62307bp)\n",
      "Seeds found: 3070\n",
      "Seeds extended: 88\n",
      "Reads with a seed: 4 (61911bp)\n",
      "Reads with an alignment: 4 (8003bp)\n",
      "Alignments: 59 (8003bp) (29 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 46.arp-v2.chrX#52660456_52942675.wd/46.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 46.arp-v2.chrX#52660456_52942675.wd/gene_alignments/aligned_pc_46.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 3 (59977bp)\n",
      "Seeds found: 2693\n",
      "Seeds extended: 87\n",
      "Reads with a seed: 3 (59977bp)\n",
      "Reads with an alignment: 3 (7702bp)\n",
      "Alignments: 58 (7702bp) (29 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 42.arp-v2.chrX#6091130_7718796.wd/gene_alignments/aligned_all_42.gaf\n",
      "Align\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 47.arp-v2.chrX#152295493_152371389.wd/47.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 47.arp-v2.chrX#152295493_152371389.wd/gene_alignments/aligned_all_47.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 47.arp-v2.chrX#152295493_152371389.wd/47.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 47.arp-v2.chrX#152295493_152371389.wd/gene_alignments/aligned_pc_47.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Alignment finished\n",
      "Input reads: 10 (720551bp)\n",
      "Seeds found: 68471\n",
      "Seeds extended: 3243\n",
      "Reads with a seed: 9 (720473bp)\n",
      "Reads with an alignment: 9 (173551bp)\n",
      "Alignments: 2381 (174543bp) (862 additional alignments discarded)\n",
      "End-to-end alignments: 1 (2964bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 42.arp-v2.chrX#6091130_7718796.wd/42.1.arp-v2.gfa\n",
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 42.arp-v2.chrX#6091130_7718796.wd/gene_alignments/aligned_pc_42.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 4 (680745bp)\n",
      "Seeds found: 62759\n",
      "Seeds extended: 3057\n",
      "Reads with a seed: 4 (680745bp)\n",
      "Reads with an alignment: 4 (162146bp)\n",
      "Alignments: 2264 (162419bp) (793 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 33.arp-v2.chr17#31725057_32882176.wd/33.1.arp-v2.gfa\n",
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 7.arp-v2.chr17#46489484_47078704.wd/7.1.arp-v2.gfa\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 33.arp-v2.chr17#31725057_32882176.wd/gene_alignments/aligned_all_33.gaf\n",
      "Align\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Load graph from 37.arp-v2.chr17#37444976_38843267.wd/37.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 7.arp-v2.chr17#46489484_47078704.wd/gene_alignments/aligned_all_7.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 37 (767360bp)\n",
      "Seeds found: 168008\n",
      "Seeds extended: 2709\n",
      "Reads with a seed: 28 (765917bp)\n",
      "Reads with an alignment: 28 (178224bp)\n",
      "Alignments: 1809 (181365bp) (900 additional alignments discarded)\n",
      "End-to-end alignments: 1 (833bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 33.arp-v2.chr17#31725057_32882176.wd/33.1.arp-v2.gfa\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 37.arp-v2.chr17#37444976_38843267.wd/gene_alignments/aligned_all_37.gaf\n",
      "Align\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment finished\n",
      "Input reads: 12 (295581bp)\n",
      "Seeds found: 19385\n",
      "Seeds extended: 886\n",
      "Reads with a seed: 11 (295136bp)\n",
      "Reads with an alignment: 11 (37891bp)\n",
      "Alignments: 577 (39447bp) (309 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Build alignment graph\n",
      "Load graph from 7.arp-v2.chr17#46489484_47078704.wd/7.1.arp-v2.gfa\n",
      "Build minimizer seeder from the graph\n",
      "Build alignment graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 33.arp-v2.chr17#31725057_32882176.wd/gene_alignments/aligned_pc_33.gaf\n",
      "Align\n",
      "Build minimizer seeder from the graph\n",
      "Alignment finished\n",
      "Input reads: 34 (1157250bp)\n",
      "Seeds found: 193074\n",
      "Seeds extended: 5010\n",
      "Reads with a seed: 32 (1157057bp)\n",
      "Reads with an alignment: 32 (261763bp)\n",
      "Alignments: 3562 (265151bp) (1448 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 7.arp-v2.chr17#46489484_47078704.wd/gene_alignments/aligned_pc_7.gaf\n",
      "Align\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 37.arp-v2.chr17#37444976_38843267.wd/37.1.arp-v2.gfa\n",
      "Alignment finished\n",
      "Input reads: 4 (187923bp)\n",
      "Seeds found: 12821\n",
      "Seeds extended: 524\n",
      "Reads with a seed: 3 (187478bp)\n",
      "Reads with an alignment: 3 (23826bp)\n",
      "Alignments: 363 (24187bp) (161 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment finished\n",
      "Input reads: 12 (687873bp)\n",
      "Seeds found: 137776\n",
      "Seeds extended: 2411\n",
      "Reads with a seed: 12 (687873bp)\n",
      "Reads with an alignment: 12 (157272bp)\n",
      "Alignments: 1640 (160291bp) (771 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Build minimizer seeder from the graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 30.arp-v2.chr17#14102393_15473806.wd/30.1.arp-v2.gfa\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 37.arp-v2.chr17#37444976_38843267.wd/gene_alignments/aligned_pc_37.gaf\n",
      "Align\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Alignment finished\n",
      "Input reads: 15 (892635bp)\n",
      "Seeds found: 150055\n",
      "Seeds extended: 3760\n",
      "Reads with a seed: 15 (892635bp)\n",
      "Reads with an alignment: 15 (202366bp)\n",
      "Alignments: 2691 (205186bp) (1069 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 27.arp-v2.chr17#14102393_15473806.wd/27.1.arp-v2.gfa\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 30.arp-v2.chr17#14102393_15473806.wd/gene_alignments/aligned_all_30.gaf\n",
      "Align\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment finished\n",
      "Input reads: 28 (480423bp)\n",
      "Seeds found: 42012\n",
      "Seeds extended: 2035\n",
      "Reads with a seed: 24 (479691bp)\n",
      "Reads with an alignment: 24 (108636bp)\n",
      "Alignments: 1458 (110178bp) (577 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Build alignment graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 30.arp-v2.chr17#14102393_15473806.wd/30.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n",
      "Build alignment graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 27.arp-v2.chr17#14102393_15473806.wd/gene_alignments/aligned_all_27.gaf\n",
      "Align\n",
      "Build minimizer seeder from the graph\n",
      "Alignment finished\n",
      "Input reads: 28 (480423bp)\n",
      "Seeds found: 42012\n",
      "Seeds extended: 2035\n",
      "Reads with a seed: 24 (479691bp)\n",
      "Reads with an alignment: 24 (108298bp)\n",
      "Alignments: 1457 (109840bp) (578 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 27.arp-v2.chr17#14102393_15473806.wd/27.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 30.arp-v2.chr17#14102393_15473806.wd/gene_alignments/aligned_pc_30.gaf\n",
      "Align\n",
      "Build alignment graph\n",
      "Alignment finished\n",
      "Input reads: 7 (345467bp)\n",
      "Seeds found: 24268\n",
      "Seeds extended: 1371\n",
      "Reads with a seed: 7 (345467bp)\n",
      "Reads with an alignment: 7 (69746bp)\n",
      "Alignments: 1031 (71139bp) (340 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 27.arp-v2.chr17#14102393_15473806.wd/gene_alignments/aligned_pc_27.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 7 (345467bp)\n",
      "Seeds found: 24268\n",
      "Seeds extended: 1372\n",
      "Reads with a seed: 7 (345467bp)\n",
      "Reads with an alignment: 7 (69194bp)\n",
      "Alignments: 1029 (70587bp) (343 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][1.9s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][0.9s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 39.arp-v2.chr16#3752324_3907325.wd/39.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 39.arp-v2.chr16#3752324_3907325.wd/gene_alignments/aligned_all_39.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 39.arp-v2.chr16#3752324_3907325.wd/39.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 39.arp-v2.chr16#3752324_3907325.wd/gene_alignments/aligned_pc_39.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.2s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 23.arp-v2.chr16#3490_789083.wd/23.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 23.arp-v2.chr16#3490_789083.wd/gene_alignments/aligned_all_23.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 65 (645910bp)\n",
      "Seeds found: 74160\n",
      "Seeds extended: 1349\n",
      "Reads with a seed: 57 (641927bp)\n",
      "Reads with an alignment: 57 (65408bp)\n",
      "Alignments: 1034 (67476bp) (315 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 23.arp-v2.chr16#3490_789083.wd/23.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 23.arp-v2.chr16#3490_789083.wd/gene_alignments/aligned_pc_23.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 42 (577782bp)\n",
      "Seeds found: 67126\n",
      "Seeds extended: 1212\n",
      "Reads with a seed: 40 (576151bp)\n",
      "Reads with an alignment: 40 (59517bp)\n",
      "Alignments: 933 (61143bp) (279 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 6.arp-v2.chr16#14891230_16405045.wd/6.1.arp-v2.gfa\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 38.arp-v2.chr16#21780281_22282000.wd/38.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 38.arp-v2.chr16#21780281_22282000.wd/gene_alignments/aligned_all_38.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 19 (459405bp)\n",
      "Seeds found: 72516\n",
      "Seeds extended: 1684\n",
      "Reads with a seed: 17 (459241bp)\n",
      "Reads with an alignment: 17 (90473bp)\n",
      "Alignments: 1276 (93972bp) (408 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "chm13v2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 38.arp-v2.chr16#21780281_22282000.wd/38.1.arp-v2.gfa\n",
      "Build minimizer seeder from the graph\n",
      "Build alignment graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 6.arp-v2.chr16#14891230_16405045.wd/gene_alignments/aligned_all_6.gaf\n",
      "Align\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 38.arp-v2.chr16#21780281_22282000.wd/gene_alignments/aligned_pc_38.gaf\n",
      "Align\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 5.arp-v2.chr16#14891230_16405045.wd/5.1.arp-v2.gfa\n",
      "Alignment finished\n",
      "Input reads: 8 (400922bp)\n",
      "Seeds found: 57698\n",
      "Seeds extended: 1473\n",
      "Reads with a seed: 8 (400922bp)\n",
      "Reads with an alignment: 8 (77892bp)\n",
      "Alignments: 1120 (81106bp) (353 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Alignment finished\n",
      "Input reads: 59 (1474213bp)\n",
      "Seeds found: 678365\n",
      "Seeds extended: 7603\n",
      "Reads with a seed: 49 (1473479bp)\n",
      "Reads with an alignment: 49 (392661bp)\n",
      "Alignments: 4459 (425987bp) (3144 additional alignments discarded)\n",
      "End-to-end alignments: 1 (85bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 6.arp-v2.chr16#14891230_16405045.wd/6.1.arp-v2.gfa\n",
      "Build minimizer seeder from the graph\n",
      "Build alignment graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 5.arp-v2.chr16#14891230_16405045.wd/gene_alignments/aligned_all_5.gaf\n",
      "Align\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 6.arp-v2.chr16#14891230_16405045.wd/gene_alignments/aligned_pc_6.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 59 (1474213bp)\n",
      "Seeds found: 678365\n",
      "Seeds extended: 7601\n",
      "Reads with a seed: 49 (1473479bp)\n",
      "Reads with an alignment: 49 (392743bp)\n",
      "Alignments: 4458 (425882bp) (3143 additional alignments discarded)\n",
      "End-to-end alignments: 1 (85bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 5.arp-v2.chr16#14891230_16405045.wd/5.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 4.arp-v2.chr16#29877571_30473116.wd/4.1.arp-v2.gfa\n",
      "Alignment finished\n",
      "Input reads: 14 (1077949bp)\n",
      "Seeds found: 482569\n",
      "Seeds extended: 5731\n",
      "Reads with a seed: 14 (1077949bp)\n",
      "Reads with an alignment: 14 (288630bp)\n",
      "Alignments: 3302 (316729bp) (2429 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][4.1s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 4.arp-v2.chr16#29877571_30473116.wd/gene_alignments/aligned_all_4.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 41 (377880bp)\n",
      "Seeds found: 96005\n",
      "Seeds extended: 1088\n",
      "Reads with a seed: 40 (377793bp)\n",
      "Reads with an alignment: 40 (76711bp)\n",
      "Alignments: 672 (77525bp) (416 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 4.arp-v2.chr16#29877571_30473116.wd/4.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 4.arp-v2.chr16#29877571_30473116.wd/gene_alignments/aligned_pc_4.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 30 (327652bp)\n",
      "Seeds found: 84718\n",
      "Seeds extended: 934\n",
      "Reads with a seed: 30 (327652bp)\n",
      "Reads with an alignment: 30 (65626bp)\n",
      "Alignments: 576 (66228bp) (358 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 5.arp-v2.chr16#14891230_16405045.wd/gene_alignments/aligned_pc_5.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 14 (1077949bp)\n",
      "Seeds found: 482569\n",
      "Seeds extended: 5729\n",
      "Reads with a seed: 14 (1077949bp)\n",
      "Reads with an alignment: 14 (288442bp)\n",
      "Alignments: 3301 (316043bp) (2428 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.2s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][2.3s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 2.arp-v2.chr15#71986221_73551280.wd/2.1.arp-v2.gfa\n",
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "[\u001b[33mWARN\u001b[0m][0.1s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.0s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 2.arp-v2.chr15#71986221_73551280.wd/gene_alignments/aligned_all_2.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 70 (1062770bp)\n",
      "Seeds found: 90518\n",
      "Seeds extended: 4389\n",
      "Reads with a seed: 64 (1062148bp)\n",
      "Reads with an alignment: 64 (214353bp)\n",
      "Alignments: 3218 (230229bp) (1171 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 2.arp-v2.chr15#71986221_73551280.wd/2.1.arp-v2.gfa\n",
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "[\u001b[33mWARN\u001b[0m][0.5s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 2.arp-v2.chr15#71986221_73551280.wd/gene_alignments/aligned_pc_2.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 34 (913575bp)\n",
      "Seeds found: 77444\n",
      "Seeds extended: 3773\n",
      "Reads with a seed: 34 (913575bp)\n",
      "Reads with an alignment: 34 (182638bp)\n",
      "Alignments: 2763 (204092bp) (1010 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "chm13v2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.2s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chm13v2\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 1.arp-v2.chr15#28411284_29947360.wd/1.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "father\n",
      "Build minimizer seeder from the graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 3.arp-v2.chr15#96579126_99750708.wd/3.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 1.arp-v2.chr15#28411284_29947360.wd/gene_alignments/aligned_all_1.gaf\n",
      "Align\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 26.arp-v2.chr15#21111068_25935855.wd/26.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "chm13v2\n",
      "Alignment finished\n",
      "Input reads: 27 (1012816bp)\n",
      "Seeds found: 142583\n",
      "Seeds extended: 4417\n",
      "Reads with a seed: 25 (1012210bp)\n",
      "Reads with an alignment: 25 (314804bp)\n",
      "Alignments: 3251 (317478bp) (1166 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 1.arp-v2.chr15#28411284_29947360.wd/1.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 1.arp-v2.chr15#28411284_29947360.wd/gene_alignments/aligned_pc_1.gaf\n",
      "Align\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 3.arp-v2.chr15#96579126_99750708.wd/gene_alignments/aligned_all_3.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 8 (806724bp)\n",
      "Seeds found: 112841\n",
      "Seeds extended: 3449\n",
      "Reads with a seed: 8 (806724bp)\n",
      "Reads with an alignment: 8 (249119bp)\n",
      "Alignments: 2596 (251430bp) (853 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 36.arp-v2.chr15#21111068_25935855.wd/36.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Alignment finished\n",
      "Input reads: 85 (2357801bp)\n",
      "Seeds found: 283071\n",
      "Seeds extended: 15929\n",
      "Reads with a seed: 80 (2356452bp)\n",
      "Reads with an alignment: 80 (690051bp)\n",
      "Alignments: 11902 (696246bp) (4027 additional alignments discarded)\n",
      "End-to-end alignments: 1 (677bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 3.arp-v2.chr15#96579126_99750708.wd/3.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.1s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.1s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 3.arp-v2.chr15#96579126_99750708.wd/gene_alignments/aligned_pc_3.gaf\n",
      "Align\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Alignment finished\n",
      "Input reads: 22 (1705973bp)\n",
      "Seeds found: 208970\n",
      "Seeds extended: 11278\n",
      "Reads with a seed: 22 (1705973bp)\n",
      "Reads with an alignment: 22 (505947bp)\n",
      "Alignments: 8553 (510490bp) (2725 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 26.arp-v2.chr15#21111068_25935855.wd/gene_alignments/aligned_all_26.gaf\n",
      "Align\n",
      "father\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.5s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 36.arp-v2.chr15#21111068_25935855.wd/gene_alignments/aligned_all_36.gaf\n",
      "Align\n",
      "chm13v2\n",
      "chm13v2\n",
      "Alignment finished\n",
      "Input reads: 155 (3872357bp)\n",
      "Seeds found: 1044222\n",
      "Seeds extended: 44830\n",
      "Reads with a seed: 129 (3870075bp)\n",
      "Reads with an alignment: 129 (1375120bp)\n",
      "Alignments: 32533 (1444805bp) (12297 additional alignments discarded)\n",
      "End-to-end alignments: 4 (339bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 26.arp-v2.chr15#21111068_25935855.wd/26.1.arp-v2.gfa\n",
      "chm13v2\n",
      "father\n",
      "chm13v2\n",
      "Alignment finished\n",
      "Input reads: 155 (3872357bp)\n",
      "Seeds found: 1044222\n",
      "Seeds extended: 44825\n",
      "Reads with a seed: 129 (3870075bp)\n",
      "Reads with an alignment: 129 (1375387bp)\n",
      "Alignments: 32529 (1444697bp) (12296 additional alignments discarded)\n",
      "End-to-end alignments: 4 (339bp)\n",
      "Build alignment graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 36.arp-v2.chr15#21111068_25935855.wd/36.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chm13v2\n",
      "gp001\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 26.arp-v2.chr15#21111068_25935855.wd/gene_alignments/aligned_pc_26.gaf\n",
      "Align\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 36.arp-v2.chr15#21111068_25935855.wd/gene_alignments/aligned_pc_36.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 13 (1771214bp)\n",
      "Seeds found: 458371\n",
      "Seeds extended: 20262\n",
      "Reads with a seed: 13 (1771214bp)\n",
      "Reads with an alignment: 13 (612151bp)\n",
      "Alignments: 14787 (640523bp) (5475 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "chm13v2\n",
      "Alignment finished\n",
      "Input reads: 13 (1771214bp)\n",
      "Seeds found: 458371\n",
      "Seeds extended: 20264\n",
      "Reads with a seed: 13 (1771214bp)\n",
      "Reads with an alignment: 13 (612303bp)\n",
      "Alignments: 14787 (640352bp) (5477 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.9s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chm13v2\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 35.arp-v2.chr11#44128586_46186902.wd/35.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 0.arp-v2.chr12#64657050_68231266.wd/0.1.arp-v2.gfa\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chm13v2\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 35.arp-v2.chr11#44128586_46186902.wd/gene_alignments/aligned_all_35.gaf\n",
      "Align\n",
      "Build alignment graph\n",
      "Alignment finished\n",
      "Input reads: 38 (1004063bp)\n",
      "Seeds found: 76626\n",
      "Seeds extended: 4939\n",
      "Reads with a seed: 37 (1003990bp)\n",
      "Reads with an alignment: 37 (232494bp)\n",
      "Alignments: 3661 (239431bp) (1278 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 35.arp-v2.chr11#44128586_46186902.wd/35.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp001\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 0.arp-v2.chr12#64657050_68231266.wd/gene_alignments/aligned_all_0.gaf\n",
      "Align\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 35.arp-v2.chr11#44128586_46186902.wd/gene_alignments/aligned_pc_35.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 15 (793790bp)\n",
      "Seeds found: 61447\n",
      "Seeds extended: 3936\n",
      "Reads with a seed: 15 (793790bp)\n",
      "Reads with an alignment: 15 (192046bp)\n",
      "Alignments: 2897 (197689bp) (1039 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.2s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chm13v2\n",
      "chm13v2\n",
      "gp002\n",
      "Alignment finished\n",
      "Input reads: 72 (2535309bp)\n",
      "Seeds found: 380820\n",
      "Seeds extended: 21877\n",
      "Reads with a seed: 66 (2534638bp)\n",
      "Reads with an alignment: 66 (882400bp)\n",
      "Alignments: 15890 (892169bp) (5987 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 43.arp-v2.chr11#31920604_32571026.wd/43.1.arp-v2.gfa\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 0.arp-v2.chr12#64657050_68231266.wd/0.1.arp-v2.gfa\n",
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 43.arp-v2.chr11#31920604_32571026.wd/gene_alignments/aligned_all_43.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 16 (250141bp)\n",
      "Seeds found: 19287\n",
      "Seeds extended: 790\n",
      "Reads with a seed: 13 (248804bp)\n",
      "Reads with an alignment: 13 (55997bp)\n",
      "Alignments: 495 (56141bp) (295 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 43.arp-v2.chr11#31920604_32571026.wd/43.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 43.arp-v2.chr11#31920604_32571026.wd/gene_alignments/aligned_pc_43.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 3 (91333bp)\n",
      "Seeds found: 5216\n",
      "Seeds extended: 249\n",
      "Reads with a seed: 3 (91333bp)\n",
      "Reads with an alignment: 3 (16101bp)\n",
      "Alignments: 172 (16177bp) (77 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 0.arp-v2.chr12#64657050_68231266.wd/gene_alignments/aligned_pc_0.gaf\n",
      "Align\n",
      "chm13v2\n",
      "Alignment finished\n",
      "Input reads: 15 (1621257bp)\n",
      "Seeds found: 212517\n",
      "Seeds extended: 13993\n",
      "Reads with a seed: 15 (1621257bp)\n",
      "Reads with an alignment: 15 (532072bp)\n",
      "Alignments: 10466 (536766bp) (3527 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.9s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 19.arp-v2.chr8#7833177_11501099.wd/19.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 21.arp-v2.chr8#76743643_77283136.wd/21.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 21.arp-v2.chr8#76743643_77283136.wd/gene_alignments/aligned_all_21.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 5 (305270bp)\n",
      "Seeds found: 18130\n",
      "Seeds extended: 1078\n",
      "Reads with a seed: 4 (304223bp)\n",
      "Reads with an alignment: 4 (94319bp)\n",
      "Alignments: 780 (94433bp) (298 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 21.arp-v2.chr8#76743643_77283136.wd/21.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 21.arp-v2.chr8#76743643_77283136.wd/gene_alignments/aligned_pc_21.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 22.arp-v2.chr9#149870673_150093988.wd/22.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 22.arp-v2.chr9#149870673_150093988.wd/gene_alignments/aligned_all_22.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 4 (14948bp)\n",
      "Seeds found: 408\n",
      "Seeds extended: 20\n",
      "Reads with a seed: 3 (14062bp)\n",
      "Reads with an alignment: 3 (1277bp)\n",
      "Alignments: 17 (1277bp) (3 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 22.arp-v2.chr9#149870673_150093988.wd/22.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 22.arp-v2.chr9#149870673_150093988.wd/gene_alignments/aligned_pc_22.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 18.arp-v2.chr7#74530623_75930620.wd/18.1.arp-v2.gfa\n",
      "Build minimizer seeder from the graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 44.arp-v2.chr7#74530623_75930620.wd/44.1.arp-v2.gfa\n",
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n",
      "Build alignment graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 41.arp-v2.chr7#97924409_97945534.wd/41.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 41.arp-v2.chr7#97924409_97945534.wd/gene_alignments/aligned_all_41.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 41.arp-v2.chr7#97924409_97945534.wd/41.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 18.arp-v2.chr7#74530623_75930620.wd/gene_alignments/aligned_all_18.gaf\n",
      "Align\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 41.arp-v2.chr7#97924409_97945534.wd/gene_alignments/aligned_pc_41.gaf\n",
      "Align\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 20.arp-v2.chr8#7833177_11501099.wd/20.1.arp-v2.gfa\n",
      "Build minimizer seeder from the graph\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment finished\n",
      "Input reads: 34 (669942bp)\n",
      "Seeds found: 244626\n",
      "Seeds extended: 2688\n",
      "Reads with a seed: 32 (669754bp)\n",
      "Reads with an alignment: 32 (165706bp)\n",
      "Alignments: 1692 (166805bp) (996 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 19.arp-v2.chr8#7833177_11501099.wd/gene_alignments/aligned_all_19.gaf\n",
      "Align\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 18.arp-v2.chr7#74530623_75930620.wd/18.1.arp-v2.gfa\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 44.arp-v2.chr7#74530623_75930620.wd/gene_alignments/aligned_all_44.gaf\n",
      "Align\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Alignment finished\n",
      "Input reads: 34 (669942bp)\n",
      "Seeds found: 244626\n",
      "Seeds extended: 2691\n",
      "Reads with a seed: 32 (669754bp)\n",
      "Reads with an alignment: 32 (165662bp)\n",
      "Alignments: 1695 (168040bp) (996 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 44.arp-v2.chr7#74530623_75930620.wd/44.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n",
      "Build alignment graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 18.arp-v2.chr7#74530623_75930620.wd/gene_alignments/aligned_pc_18.gaf\n",
      "Align\n",
      "Build minimizer seeder from the graph\n",
      "Alignment finished\n",
      "Input reads: 21 (645531bp)\n",
      "Seeds found: 233979\n",
      "Seeds extended: 2582\n",
      "Reads with a seed: 21 (645531bp)\n",
      "Reads with an alignment: 21 (160986bp)\n",
      "Alignments: 1626 (162914bp) (956 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 44.arp-v2.chr7#74530623_75930620.wd/gene_alignments/aligned_pc_44.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 21 (645531bp)\n",
      "Seeds found: 233979\n",
      "Seeds extended: 2581\n",
      "Reads with a seed: 21 (645531bp)\n",
      "Reads with an alignment: 21 (160707bp)\n",
      "Alignments: 1625 (162635bp) (956 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Alignment finished\n",
      "Input reads: 95 (2400299bp)\n",
      "Seeds found: 1248049\n",
      "Seeds extended: 19447\n",
      "Reads with a seed: 90 (2399869bp)\n",
      "Reads with an alignment: 90 (778111bp)\n",
      "Alignments: 14295 (797770bp) (5152 additional alignments discarded)\n",
      "End-to-end alignments: 3 (569bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 19.arp-v2.chr8#7833177_11501099.wd/19.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chm13v2\n",
      "Build alignment graph\n",
      "Build alignment graph\n",
      "chm13v2\n",
      "Build minimizer seeder from the graph\n",
      "gp003\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 20.arp-v2.chr8#7833177_11501099.wd/gene_alignments/aligned_all_20.gaf\n",
      "Align\n",
      "Build minimizer seeder from the graph\n",
      "Alignment finished\n",
      "Input reads: 95 (2400299bp)\n",
      "Seeds found: 1248049\n",
      "Seeds extended: 19443\n",
      "Reads with a seed: 90 (2399869bp)\n",
      "Reads with an alignment: 90 (779194bp)\n",
      "Alignments: 14297 (799249bp) (5146 additional alignments discarded)\n",
      "End-to-end alignments: 3 (569bp)\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 19.arp-v2.chr8#7833177_11501099.wd/gene_alignments/aligned_pc_19.gaf\n",
      "Align\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 20.arp-v2.chr8#7833177_11501099.wd/20.1.arp-v2.gfa\n",
      "gp002\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 25.arp-v2.chr5#127296578_127357028.wd/25.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 25.arp-v2.chr5#127296578_127357028.wd/gene_alignments/aligned_all_25.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 1 (60450bp)\n",
      "Seeds found: 864\n",
      "Seeds extended: 56\n",
      "Reads with a seed: 1 (60450bp)\n",
      "Reads with an alignment: 1 (2046bp)\n",
      "Alignments: 32 (2284bp) (24 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 25.arp-v2.chr5#127296578_127357028.wd/25.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 25.arp-v2.chr5#127296578_127357028.wd/gene_alignments/aligned_pc_25.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 1 (60450bp)\n",
      "Seeds found: 864\n",
      "Seeds extended: 56\n",
      "Reads with a seed: 1 (60450bp)\n",
      "Reads with an alignment: 1 (2039bp)\n",
      "Alignments: 32 (2277bp) (24 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Alignment finished\n",
      "Input reads: 23 (1834064bp)\n",
      "Seeds found: 931013\n",
      "Seeds extended: 14536\n",
      "Reads with a seed: 23 (1834064bp)\n",
      "Reads with an alignment: 23 (570806bp)\n",
      "Alignments: 11069 (583883bp) (3467 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 29.arp-v2.chr5#113218068_113356772.wd/29.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 29.arp-v2.chr5#113218068_113356772.wd/gene_alignments/aligned_all_29.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 2 (924bp)\n",
      "Seeds found: 1\n",
      "Seeds extended: 1\n",
      "Reads with a seed: 1 (817bp)\n",
      "Reads with an alignment: 1 (16bp)\n",
      "Alignments: 1 (16bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 29.arp-v2.chr5#113218068_113356772.wd/29.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 29.arp-v2.chr5#113218068_113356772.wd/gene_alignments/aligned_pc_29.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 40.arp-v2.chr5#176841220_178167392.wd/40.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 40.arp-v2.chr5#176841220_178167392.wd/gene_alignments/aligned_all_40.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 49 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 40.arp-v2.chr5#176841220_178167392.wd/40.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 40.arp-v2.chr5#176841220_178167392.wd/gene_alignments/aligned_pc_40.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 35 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 20.arp-v2.chr8#7833177_11501099.wd/gene_alignments/aligned_pc_20.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 23 (1834064bp)\n",
      "Seeds found: 931013\n",
      "Seeds extended: 14537\n",
      "Reads with a seed: 23 (1834064bp)\n",
      "Reads with an alignment: 23 (570549bp)\n",
      "Alignments: 11067 (583591bp) (3470 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "gp004\n",
      "chm13v2\n",
      "chm13v2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.2s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "[\u001b[33mWARN\u001b[0m][1.5s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 17.arp-v2.chr3#198719061_200350356.wd/17.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "gp001\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 17.arp-v2.chr3#198719061_200350356.wd/gene_alignments/aligned_all_17.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 55 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 17.arp-v2.chr3#198719061_200350356.wd/17.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp003\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 17.arp-v2.chr3#198719061_200350356.wd/gene_alignments/aligned_pc_17.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 23 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "gp005\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.1s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 45.arp-v2.chr4#1564721_2107304.wd/45.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 45.arp-v2.chr4#1564721_2107304.wd/gene_alignments/aligned_all_45.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 18 (420245bp)\n",
      "Seeds found: 28118\n",
      "Seeds extended: 798\n",
      "Reads with a seed: 15 (419964bp)\n",
      "Reads with an alignment: 15 (57993bp)\n",
      "Alignments: 539 (58499bp) (259 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 45.arp-v2.chr4#1564721_2107304.wd/45.1.arp-v2.gfa\n",
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.0s] excluding all Walks; including them requires --connected (and connectivity index)\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 45.arp-v2.chr4#1564721_2107304.wd/gene_alignments/aligned_pc_45.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 11 (408517bp)\n",
      "Seeds found: 27647\n",
      "Seeds extended: 786\n",
      "Reads with a seed: 11 (408517bp)\n",
      "Reads with an alignment: 11 (57741bp)\n",
      "Alignments: 528 (58247bp) (258 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "chm13v2\n",
      "chm13v2\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 16.arp-v2.chr3#198719061_200350356.wd/16.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.5s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp006\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 16.arp-v2.chr3#198719061_200350356.wd/gene_alignments/aligned_all_16.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 55 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 16.arp-v2.chr3#198719061_200350356.wd/16.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "chm13v2\n",
      "Build minimizer seeder from the graph\n",
      "gp004\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 16.arp-v2.chr3#198719061_200350356.wd/gene_alignments/aligned_pc_16.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 23 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.4s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "father\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[\u001b[33mWARN\u001b[0m][0.3s] excluding all Walks; including them requires --connected (and connectivity index)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gp007\n",
      "chm13v2\n",
      "chm13v2\n",
      "gp005\n",
      "chm13v2\n",
      "gp008\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 15.arp-v2.chr2#239539540_239891533.wd/15.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 15.arp-v2.chr2#239539540_239891533.wd/gene_alignments/aligned_all_15.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 7 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 15.arp-v2.chr2#239539540_239891533.wd/15.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 15.arp-v2.chr2#239539540_239891533.wd/gene_alignments/aligned_pc_15.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 0 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "chm13v2\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 14.arp-v2.chr2#196544429_204824491.wd/14.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chm13v2\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 14.arp-v2.chr2#196544429_204824491.wd/gene_alignments/aligned_all_14.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 183 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 14.arp-v2.chr2#196544429_204824491.wd/14.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "father\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 10.arp-v2.chr1#144803785_145165379.wd/10.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 10.arp-v2.chr1#144803785_145165379.wd/gene_alignments/aligned_all_10.gaf\n",
      "Align\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment finished\n",
      "Input reads: 22 (303298bp)\n",
      "Seeds found: 38346\n",
      "Seeds extended: 573\n",
      "Reads with a seed: 17 (302950bp)\n",
      "Reads with an alignment: 17 (57374bp)\n",
      "Alignments: 396 (57463bp) (177 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 10.arp-v2.chr1#144803785_145165379.wd/10.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 13.arp-v2.chr2#44188661_44367747.wd/13.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n",
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 10.arp-v2.chr1#144803785_145165379.wd/gene_alignments/aligned_pc_10.gaf\n",
      "Align\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 13.arp-v2.chr2#44188661_44367747.wd/gene_alignments/aligned_all_13.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 14 (279936bp)\n",
      "Seeds found: 36792\n",
      "Seeds extended: 538\n",
      "Reads with a seed: 14 (279936bp)\n",
      "Reads with an alignment: 14 (53545bp)\n",
      "Alignments: 371 (53634bp) (167 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Alignment finished\n",
      "Input reads: 4 (91862bp)\n",
      "Seeds found: 4393\n",
      "Seeds extended: 123\n",
      "Reads with a seed: 4 (91862bp)\n",
      "Reads with an alignment: 4 (10253bp)\n",
      "Alignments: 65 (10253bp) (58 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 13.arp-v2.chr2#44188661_44367747.wd/13.1.arp-v2.gfa\n",
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Build minimizer seeder from the graph\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 13.arp-v2.chr2#44188661_44367747.wd/gene_alignments/aligned_pc_13.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 2 (91145bp)\n",
      "Seeds found: 4387\n",
      "Seeds extended: 119\n",
      "Reads with a seed: 2 (91145bp)\n",
      "Reads with an alignment: 2 (10048bp)\n",
      "Alignments: 62 (10048bp) (57 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 14.arp-v2.chr2#196544429_204824491.wd/gene_alignments/aligned_pc_14.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 60 (0bp)\n",
      "Seeds found: 0\n",
      "Seeds extended: 0\n",
      "Reads with a seed: 0 (0bp)\n",
      "Reads with an alignment: 0 (0bp)\n",
      "Alignments: 0 (0bp)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "gp006\n",
      "father\n",
      "gp009\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 12.arp-v2.chr2#59063565_61598724.wd/12.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 12.arp-v2.chr2#59063565_61598724.wd/gene_alignments/aligned_all_12.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 38 (1116813bp)\n",
      "Seeds found: 61901\n",
      "Seeds extended: 3753\n",
      "Reads with a seed: 28 (1114599bp)\n",
      "Reads with an alignment: 28 (124938bp)\n",
      "Alignments: 3410 (125241bp) (343 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 12.arp-v2.chr2#59063565_61598724.wd/12.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 12.arp-v2.chr2#59063565_61598724.wd/gene_alignments/aligned_pc_12.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 10 (785322bp)\n",
      "Seeds found: 61131\n",
      "Seeds extended: 3030\n",
      "Reads with a seed: 10 (785322bp)\n",
      "Reads with an alignment: 10 (111457bp)\n",
      "Alignments: 2704 (111896bp) (326 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 9.arp-v2.chr1#146396099_147746605.wd/9.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 9.arp-v2.chr1#146396099_147746605.wd/gene_alignments/aligned_all_9.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 63 (949494bp)\n",
      "Seeds found: 116806\n",
      "Seeds extended: 3452\n",
      "Reads with a seed: 42 (946433bp)\n",
      "Reads with an alignment: 42 (227545bp)\n",
      "Alignments: 2635 (229573bp) (817 additional alignments discarded)\n",
      "End-to-end alignments: 1 (164bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 9.arp-v2.chr1#146396099_147746605.wd/9.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 9.arp-v2.chr1#146396099_147746605.wd/gene_alignments/aligned_pc_9.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 9 (453656bp)\n",
      "Seeds found: 67958\n",
      "Seeds extended: 1536\n",
      "Reads with a seed: 9 (453656bp)\n",
      "Reads with an alignment: 9 (120787bp)\n",
      "Alignments: 1179 (121373bp) (357 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "gp007\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 8.arp-v2.chr1#146396099_147746605.wd/8.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 8.arp-v2.chr1#146396099_147746605.wd/gene_alignments/aligned_all_8.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 63 (949494bp)\n",
      "Seeds found: 116806\n",
      "Seeds extended: 3451\n",
      "Reads with a seed: 42 (946433bp)\n",
      "Reads with an alignment: 42 (227817bp)\n",
      "Alignments: 2637 (230281bp) (814 additional alignments discarded)\n",
      "End-to-end alignments: 1 (164bp)\n",
      "GraphAligner bioconda 1.0.13-\n",
      "Load graph from 8.arp-v2.chr1#146396099_147746605.wd/8.1.arp-v2.gfa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GraphAligner bioconda 1.0.13-\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Build alignment graph\n",
      "Build minimizer seeder from the graph\n",
      "Minimizer seeds, length 15, window size 20, density 10\n",
      "Seed cluster size 1\n",
      "Alignment bandwidth 10\n",
      "Clip alignment ends with identity < 66%\n",
      "X-drop DP score cutoff 14705\n",
      "write alignments to 8.arp-v2.chr1#146396099_147746605.wd/gene_alignments/aligned_pc_8.gaf\n",
      "Align\n",
      "Alignment finished\n",
      "Input reads: 9 (453656bp)\n",
      "Seeds found: 67958\n",
      "Seeds extended: 1535\n",
      "Reads with a seed: 9 (453656bp)\n",
      "Reads with an alignment: 9 (121066bp)\n",
      "Alignments: 1181 (122088bp) (354 additional alignments discarded)\n",
      "End-to-end alignments: 0 (0bp)\n",
      "gp010\n",
      "gp001\n",
      "gp002\n",
      "chm13v2\n",
      "gp011\n",
      "gp008\n",
      "father\n",
      "gp002\n",
      "gp012\n",
      "gp003\n",
      "gp009\n",
      "gp013\n",
      "gp001\n",
      "gp001\n",
      "gp003\n",
      "gp004\n",
      "gp014\n",
      "gp010\n",
      "gp005\n",
      "chm13v2\n",
      "chm13v2\n",
      "chm13v2\n",
      "gp011\n",
      "father\n",
      "gp015\n",
      "chm13v2\n",
      "gp006\n",
      "chm13v2\n",
      "father\n",
      "father\n",
      "father\n",
      "gp012\n",
      "gp016\n",
      "father\n",
      "gp002\n",
      "gp002\n",
      "gp007\n",
      "gp013\n",
      "gp004\n",
      "gp017\n",
      "gp008\n",
      "gp014\n",
      "gp018\n",
      "gp009\n",
      "father\n",
      "gp001\n",
      "gp019\n",
      "gp015\n",
      "gp010\n",
      "gp020\n",
      "gp016\n",
      "gp011\n",
      "gp003\n",
      "gp021\n",
      "gp005\n",
      "gp001\n",
      "gp003\n",
      "gp012\n",
      "gp017\n",
      "gp002\n",
      "gp022\n",
      "father\n",
      "gp013\n",
      "gp018\n",
      "gp023\n",
      "gp006\n",
      "gp024\n",
      "gp014\n",
      "gp019\n",
      "gp025\n",
      "father\n",
      "father\n",
      "gp015\n",
      "gp026\n",
      "gp004\n",
      "gp003\n",
      "gp020\n",
      "gp027\n",
      "gp016\n",
      "gp029\n",
      "gp021\n",
      "gp017\n",
      "gp030\n",
      "gp018\n",
      "gp007\n",
      "gp004\n",
      "gp022\n",
      "gp004\n",
      "gp031\n",
      "gp019\n",
      "father\n",
      "gp005\n",
      "gp032\n",
      "gp023\n",
      "gp020\n",
      "gp033\n",
      "gp024\n",
      "gp034\n",
      "gp021\n",
      "gp005\n",
      "gp025\n",
      "gp035\n",
      "gp026\n",
      "gp022\n",
      "gp002\n",
      "gp027\n",
      "gp008\n",
      "gp036\n",
      "gp006\n",
      "gp023\n",
      "father\n",
      "gp037\n",
      "gp006\n",
      "gp001\n",
      "gp001\n",
      "gp024\n",
      "gp029\n",
      "gp005\n",
      "gp038\n",
      "father\n",
      "gp039\n",
      "gp025\n",
      "gp001\n",
      "gp009\n",
      "gp030\n",
      "gp040\n",
      "father\n",
      "gp026\n",
      "gp007\n",
      "gp041\n",
      "gp031\n",
      "gp007\n",
      "hg38_no_alt\n",
      "gp027\n",
      "mother\n",
      "gp032\n",
      "gp029\n",
      "son\n",
      "gp033\n",
      "gp001\n",
      "gp008\n",
      "gp006\n",
      "gp030\n",
      "gp034\n",
      "gp010\n",
      "gp031\n",
      "father\n",
      "gp001\n",
      "gp035\n",
      "gp008\n",
      "gp032\n",
      "gp009\n",
      "gp036\n",
      "gp033\n",
      "gp003\n",
      "gp037\n",
      "father\n",
      "gp034\n",
      "gp038\n",
      "gp035\n",
      "gp011\n",
      "gp001\n",
      "gp039\n",
      "gp010\n",
      "gp009\n",
      "gp036\n",
      "gp007\n",
      "gp040\n",
      "father\n",
      "father\n",
      "gp037\n",
      "father\n",
      "gp041\n",
      "gp038\n",
      "gp002\n",
      "gp011\n",
      "hg38_no_alt\n",
      "gp012\n",
      "gp039\n",
      "mother\n",
      "father\n",
      "gp002\n",
      "gp001\n",
      "gp010\n",
      "father\n",
      "gp040\n",
      "son\n",
      "gp001\n",
      "gp012\n",
      "gp041\n",
      "gp002\n",
      "gp008\n",
      "hg38_no_alt\n",
      "gp013\n",
      "mother\n",
      "son\n",
      "gp011\n",
      "gp013\n",
      "gp004\n",
      "gp001\n",
      "gp014\n",
      "gp014\n",
      "gp012\n",
      "gp009\n",
      "gp002\n",
      "gp015\n",
      "gp015\n",
      "gp003\n",
      "gp002\n",
      "gp016\n",
      "gp013\n",
      "gp003\n",
      "gp016\n",
      "gp010\n",
      "gp017\n",
      "gp005\n",
      "gp014\n",
      "gp002\n",
      "gp017\n",
      "gp018\n",
      "gp003\n",
      "gp019\n",
      "gp015\n",
      "gp018\n",
      "gp002\n",
      "gp011\n",
      "gp002\n",
      "gp020\n",
      "gp016\n",
      "gp019\n",
      "gp021\n",
      "gp003\n",
      "gp006\n",
      "gp012\n",
      "gp001\n",
      "gp022\n",
      "gp004\n",
      "gp017\n",
      "gp020\n",
      "gp001\n",
      "gp001\n",
      "gp004\n",
      "gp023\n",
      "gp003\n",
      "gp018\n",
      "gp024\n",
      "gp013\n",
      "gp004\n",
      "gp021\n",
      "gp003\n",
      "gp025\n",
      "gp019\n",
      "gp007\n",
      "father\n",
      "gp026\n",
      "gp022\n",
      "gp014\n",
      "gp020\n",
      "gp001\n",
      "gp027\n",
      "gp003\n",
      "gp003\n",
      "gp004\n",
      "gp023\n",
      "gp001\n",
      "gp021\n",
      "gp029\n",
      "gp005\n",
      "gp008\n",
      "gp015\n",
      "gp002\n",
      "gp030\n",
      "gp005\n",
      "gp022\n",
      "gp024\n",
      "gp031\n",
      "gp005\n",
      "gp016\n",
      "gp001\n",
      "gp023\n",
      "gp025\n",
      "gp001\n",
      "gp004\n",
      "gp032\n",
      "gp033\n",
      "gp026\n",
      "gp024\n",
      "gp004\n",
      "gp017\n",
      "gp009\n",
      "gp005\n",
      "gp001\n",
      "gp001\n",
      "gp034\n",
      "gp027\n",
      "gp025\n",
      "gp006\n",
      "gp006\n",
      "gp035\n",
      "gp029\n",
      "gp018\n",
      "gp026\n",
      "gp036\n",
      "gp001\n",
      "gp002\n",
      "gp037\n",
      "gp006\n",
      "gp004\n",
      "gp027\n",
      "gp030\n",
      "gp010\n",
      "father\n",
      "gp004\n",
      "gp019\n",
      "gp038\n",
      "gp005\n",
      "gp029\n",
      "gp002\n",
      "gp039\n",
      "gp031\n",
      "gp006\n",
      "gp005\n",
      "gp040\n",
      "gp002\n",
      "gp030\n",
      "gp020\n",
      "gp007\n",
      "gp032\n",
      "gp007\n",
      "gp041\n",
      "gp011\n",
      "gp031\n",
      "hg38_no_alt\n",
      "gp033\n",
      "gp021\n",
      "mother\n",
      "gp032\n",
      "gp007\n",
      "son\n",
      "gp034\n",
      "gp033\n",
      "gp022\n",
      "gp008\n",
      "gp005\n",
      "gp003\n",
      "gp012\n",
      "gp007\n",
      "gp035\n",
      "gp006\n",
      "gp034\n",
      "gp008\n",
      "gp002\n",
      "gp023\n",
      "gp006\n",
      "gp036\n",
      "gp035\n",
      "gp002\n",
      "gp005\n",
      "gp036\n",
      "father\n",
      "gp037\n",
      "gp024\n",
      "gp013\n",
      "gp008\n",
      "gp003\n",
      "gp037\n",
      "gp002\n",
      "gp038\n",
      "gp002\n",
      "gp009\n",
      "gp008\n",
      "gp025\n",
      "gp038\n",
      "gp039\n",
      "gp009\n",
      "gp006\n",
      "gp039\n",
      "gp014\n",
      "gp007\n",
      "gp040\n",
      "gp026\n",
      "gp007\n",
      "gp040\n",
      "gp004\n",
      "gp003\n",
      "gp041\n",
      "gp002\n",
      "gp006\n",
      "gp009\n",
      "gp041\n",
      "gp002\n",
      "gp027\n",
      "gp003\n",
      "gp015\n",
      "hg38_no_alt\n",
      "gp010\n",
      "hg38_no_alt\n",
      "mother\n",
      "gp009\n",
      "father\n",
      "father\n",
      "mother\n",
      "gp029\n",
      "gp010\n",
      "son\n",
      "son\n",
      "gp002\n",
      "gp007\n",
      "gp030\n",
      "gp008\n",
      "gp016\n",
      "gp008\n",
      "gp010\n",
      "gp031\n",
      "gp011\n",
      "gp010\n",
      "gp003\n",
      "gp017\n",
      "gp032\n",
      "gp011\n",
      "gp007\n",
      "gp004\n",
      "gp003\n",
      "gp033\n",
      "father\n",
      "father\n",
      "gp008\n",
      "gp018\n",
      "gp009\n",
      "gp011\n",
      "gp001\n",
      "gp009\n",
      "gp034\n",
      "gp003\n",
      "gp011\n",
      "gp005\n",
      "gp012\n",
      "gp003\n",
      "gp004\n",
      "gp035\n",
      "gp012\n",
      "gp019\n",
      "gp004\n",
      "gp036\n",
      "gp008\n",
      "gp012\n",
      "gp037\n",
      "gp020\n",
      "gp012\n",
      "gp010\n",
      "gp009\n",
      "gp013\n",
      "gp010\n",
      "gp003\n",
      "gp038\n",
      "gp003\n",
      "gp013\n",
      "gp021\n",
      "gp006\n",
      "gp039\n",
      "gp013\n",
      "gp013\n",
      "gp004\n",
      "gp040\n",
      "gp014\n",
      "gp022\n",
      "gp005\n",
      "gp003\n",
      "gp011\n",
      "gp011\n",
      "gp009\n",
      "gp010\n",
      "gp041\n",
      "gp014\n",
      "gp004\n",
      "hg38_no_alt\n",
      "gp014\n",
      "gp023\n",
      "gp005\n",
      "gp014\n",
      "mother\n",
      "gp004\n",
      "gp007\n",
      "gp005\n",
      "gp015\n",
      "gp004\n",
      "son\n",
      "gp011\n",
      "gp024\n",
      "gp012\n",
      "gp012\n",
      "gp015\n",
      "gp015\n",
      "gp015\n",
      "gp010\n",
      "gp016\n",
      "gp025\n",
      "gp016\n",
      "gp001\n",
      "gp006\n",
      "gp013\n",
      "gp013\n",
      "gp026\n",
      "gp005\n",
      "gp016\n",
      "gp004\n",
      "gp016\n",
      "gp004\n",
      "gp017\n",
      "gp012\n",
      "gp027\n",
      "gp017\n",
      "gp006\n",
      "gp011\n",
      "gp005\n",
      "gp008\n",
      "gp017\n",
      "gp006\n",
      "gp017\n",
      "gp014\n",
      "gp014\n",
      "gp029\n",
      "gp018\n",
      "gp004\n",
      "gp005\n",
      "gp005\n",
      "gp018\n",
      "gp030\n",
      "gp013\n",
      "gp002\n",
      "gp018\n",
      "gp018\n",
      "gp007\n",
      "gp015\n",
      "gp015\n",
      "gp019\n",
      "gp031\n",
      "gp012\n",
      "gp019\n",
      "gp006\n",
      "gp032\n",
      "gp019\n",
      "gp019\n",
      "gp014\n",
      "gp007\n",
      "gp009\n",
      "gp020\n",
      "gp016\n",
      "gp016\n",
      "gp033\n",
      "gp020\n",
      "gp007\n",
      "gp005\n",
      "gp005\n",
      "gp006\n",
      "gp020\n",
      "gp020\n",
      "gp013\n",
      "gp034\n",
      "gp008\n",
      "gp006\n",
      "gp021\n",
      "gp015\n",
      "gp006\n",
      "gp017\n",
      "gp021\n",
      "gp017\n",
      "gp035\n",
      "gp021\n",
      "gp021\n",
      "gp005\n",
      "gp036\n",
      "gp022\n",
      "gp022\n",
      "gp014\n",
      "gp016\n",
      "gp018\n",
      "gp010\n",
      "gp001\n",
      "gp007\n",
      "gp022\n",
      "gp018\n",
      "gp008\n",
      "gp022\n",
      "gp037\n",
      "gp008\n",
      "gp023\n",
      "gp038\n",
      "gp023\n",
      "gp023\n",
      "gp007\n",
      "gp019\n",
      "gp017\n",
      "gp023\n",
      "gp009\n",
      "gp019\n",
      "gp006\n",
      "gp006\n",
      "gp039\n",
      "gp011\n",
      "gp015\n",
      "gp007\n",
      "gp024\n",
      "gp007\n",
      "gp024\n",
      "gp024\n",
      "gp040\n",
      "gp024\n",
      "gp020\n",
      "gp020\n",
      "gp041\n",
      "gp009\n",
      "gp016\n",
      "gp008\n",
      "gp025\n",
      "gp018\n",
      "gp025\n",
      "gp003\n",
      "gp009\n",
      "gp025\n",
      "gp025\n",
      "gp006\n",
      "hg38_no_alt\n",
      "gp002\n",
      "gp021\n",
      "mother\n",
      "gp021\n",
      "gp001\n",
      "gp010\n",
      "gp001\n",
      "gp026\n",
      "gp026\n",
      "gp012\n",
      "gp008\n",
      "gp026\n",
      "gp026\n",
      "son\n",
      "gp019\n",
      "gp017\n",
      "gp022\n",
      "gp008\n",
      "gp007\n",
      "gp007\n",
      "gp008\n",
      "gp027\n",
      "gp022\n",
      "gp027\n",
      "gp027\n",
      "gp027\n",
      "gp001\n",
      "gp010\n",
      "gp001\n",
      "gp013\n",
      "gp020\n",
      "gp010\n",
      "gp009\n",
      "gp023\n",
      "gp029\n",
      "gp029\n",
      "gp018\n",
      "gp029\n",
      "gp023\n",
      "gp029\n",
      "gp011\n",
      "gp030\n",
      "gp030\n",
      "gp021\n",
      "gp024\n",
      "gp014\n",
      "gp007\n",
      "gp030\n",
      "gp030\n",
      "gp009\n",
      "gp024\n",
      "gp019\n",
      "gp031\n",
      "gp009\n",
      "gp031\n",
      "gp009\n",
      "gp011\n",
      "gp022\n",
      "gp031\n",
      "gp031\n",
      "gp025\n",
      "gp008\n",
      "gp008\n",
      "gp010\n",
      "gp011\n",
      "gp025\n",
      "gp032\n",
      "gp032\n",
      "gp032\n",
      "gp032\n",
      "gp023\n",
      "gp012\n",
      "gp020\n",
      "gp026\n",
      "gp015\n",
      "gp026\n",
      "gp033\n",
      "gp010\n",
      "gp004\n",
      "gp033\n",
      "gp033\n",
      "gp033\n",
      "gp024\n",
      "gp027\n",
      "gp012\n",
      "gp034\n",
      "gp027\n",
      "gp010\n",
      "gp008\n",
      "gp012\n",
      "gp021\n",
      "gp034\n",
      "gp010\n",
      "gp034\n",
      "gp011\n",
      "gp034\n",
      "gp025\n",
      "gp035\n",
      "gp029\n",
      "gp035\n",
      "gp009\n",
      "gp013\n",
      "gp009\n",
      "gp016\n",
      "gp035\n",
      "gp035\n",
      "gp029\n",
      "gp036\n",
      "gp022\n",
      "gp026\n",
      "gp011\n",
      "gp003\n",
      "gp036\n",
      "gp030\n",
      "gp036\n",
      "gp036\n",
      "gp013\n",
      "gp030\n",
      "gp013\n",
      "gp037\n",
      "gp002\n",
      "gp037\n",
      "gp012\n",
      "gp037\n",
      "gp011\n",
      "gp037\n",
      "gp031\n",
      "gp011\n",
      "gp023\n",
      "gp017\n",
      "gp027\n",
      "gp031\n",
      "gp038\n",
      "gp014\n",
      "gp009\n"
     ]
    }
   ],
   "source": [
    "with Pool() as p:\n",
    "    p.starmap(produce_plotable, tasks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
